src/hooks/spaces/useSpaceBySlug.ts

```diff
@@ -19,48 +24,70 @@ export type UseSpaceBySlugResult = {
  * @param spaceSlug
  * @returns
  */
-export const useSpaceBySlug = (spaceSlug?: string): UseSpaceBySlugResult => {
+export const useSpaceBySlug = (
+  worldSlug?: WorldSlug,
```

Comment:
Spaces can be found by `worldId+spaceSlug` combo and if there is luck, maybe just `spaceSlug` will return a single, not multiple result.

The idea behind having multiple hooks is giving the components the power to determine what data they need and the ability to reduce DB queries.

Looking at this hook, it might be more viable to leave it as such and add 1 or 2 more that deal with the different combos.  This might just boil down to the peculiarities of hooks that don't allow branching out like in https://github.com/sparkletown/sparkle/blob/staging/src/api/space.ts#L34 (`findSpaceBySlug` calling `findWorldBySlug`) so it's left to components and how they are structured to provide the logic so it's easier to combine simpler hooks.

```suggestion
  worldId?: WorldId,
```

Reply:
I want to revisit all of this as part of "higher up the stack" should deal with this. I don't think components should be going off and finding a world or a space. They should be given it either as a prop or as context. Components having to decide how to deal with the database efficiently is breaking separation of concerns in my opinion.